main.tf

```diff
@@ -1,11 +1,39 @@
+data "aws_iam_policy_document" "this" {
+  # dont create this if no services or iam arns are specified
+  count = length(concat(var.allow_publish_aws_services, var.allow_publish_iam_arns)) > 0 ? 1 : 0
+  statement {
+    effect    = "Allow"
+    actions   = ["sns:Publish"]
+    resources = ["*"]
+
+    dynamic principals {
+      for_each = length(var.allow_publish_aws_services) > 0 ? ["_enable"] : []
+      content {
+        type        = "Service"
+        identifiers = var.allow_publish_aws_services
+      }
+    }
+
+    dynamic principals {
```

Comment:
Wrap names (`principals`) with double-quotes.

Reply:
Roger that, done!